### PR TITLE
38159 Sending demographics confirmation only if verified from upstream

### PR DESCRIPTION
**Pull request by kanchanasuriya**
## Description of change

Fix for demographics confirmation payload accidentally sending `...NeedsUpdate` to true when confirmation pages are not displayed.  Also, included new CHIP endpoints in Statsd for measuring the metrics.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#38159

## Things to know about this PR

- [x] rubocop
- [x] rspec

### CODE REVIEW

**Review by ericboehs — DISMISSED** [review]

File: modules/check_in/app/services/v2/chip/service.rb [hunk 1]
```diff
@@ -210,6 +214,29 @@ def appointment_identifiers
...
+        end
+        unless next_of_kin_up_to_date.nil?
+          hash[:demographicConfirmations].merge!({ nextOfKinNeedsUpdate: !next_of_kin_up_to_date,
+                                                   nextOfKinConfirmedAt: confirmed_at })
+        end
+        unless emergency_contact_up_to_date.nil?
+          hash[:demographicConfirmations].merge!({ emergencyContactNeedsUpdate: !emergency_contact_up_to_date,
+                                                   emergencyContactConfirmedAt: confirmed_at })
+        end
+        hash
```
Comment (randomsync) [inline modules/check_in/app/services/v2/chip/service.rb:238]: This is kinda hard for me to follow. I'd suggest:
```
        hsh = {}

        unless check_in_body[:demographics_up_to_date].nil?
          hsh[:demographicsNeedsUpdate] = check_in_body[:demographics_up_to_date] ? false : true
          hsh[:demographicsConfirmedAt] = confirmed_at
        end

        unless check_in_body[:next_of_kin_up_to_date].nil?
          hsh[:nextOfKinNeedsUpdate] = check_in_body[:next_of_kin_up_to_date] ? false : true
          hsh[:nextOfKinConfirmedAt] = confirmed_at
        end

        unless check_in_body[:emergency_contact_up_to_date].nil?
          hsh[:emergencyContactNeedsUpdate] = check_in_body[:emergency_contact_up_to_date] ? false : true
          hsh[:emergencyContactConfirmedAt] = confirmed_at
        end

        {
          demographicConfirmations: hsh
        }
```

Maybe then there's no need to have an extra private method either.

File: modules/check_in/spec/services/v2/chip/service_spec.rb [hunk 2]
```diff
@@ -337,4 +340,185 @@
...
+        end
+
+        context 'when no demographics data available in check_in_body' do
+          let(:demographics_confirmation_hash) do
+            {
+              demographicConfirmations: {}
+            }
+          end
+
+          it 'returns demographics confirmation hash with all demographics data' do
```
Comment (randomsync) [inline modules/check_in/spec/services/v2/chip/service_spec.rb:454]: Shouldn't this say "... with **no** demographics data"?

File: modules/check_in/spec/services/v2/chip/service_spec.rb [hunk 3]
```diff
@@ -337,4 +340,185 @@
...
+                emergencyContactConfirmedAt: Time.zone.now.iso8601
+              }
+            }
+          end
+
+          it 'returns demographics confirmation hash with all demographics data' do
+            expect(subject.build(check_in: valid_check_in, params: params)
+                          .demographic_confirmations).to eq(demographics_confirmation_hash)
+          end
+        end
```
Comment (randomsync) [inline modules/check_in/spec/services/v2/chip/service_spec.rb:520]: The tests look good. I would suggest adding a test for when the flag is turned off and no demographic confirmation parameters are sent, as that is the most likely real life scenario.

**Review by randomsync — CHANGES_REQUESTED** [review]

File: modules/check_in/spec/services/v2/chip/service_spec.rb (hunk 2 above)
Comment (kanchanasuriya) [reply modules/check_in/spec/services/v2/chip/service_spec.rb]: Thank you, copy-paste issue.  Fixed it [604e461](https://github.com/department-of-veterans-affairs/vets-api/pull/9336/commits/604e4613bd3bd2d414058dde830ed8cb55b7688f)

File: modules/check_in/spec/services/v2/chip/service_spec.rb (hunk 3 above)
Comment (kanchanasuriya) [reply modules/check_in/spec/services/v2/chip/service_spec.rb:520]: Basically, that's the issue getting fixed in this change.  Currently, if no demographic confirmation parameters are sent, the vets-api still sends `...NeedsUpdate` with true to the downstream.  

Highlighted test here covers the case where `next_of_kin_up_to_date` & `emergency_contact_up_to_date` not sent, but demographics_confirmation_hash sends a payload with `true` in `next_of_kin_up_to_date` & `emergency_contact_up_to_date`

File: modules/check_in/app/services/v2/chip/service.rb (hunk 1 above)
Comment (kanchanasuriya) [reply modules/check_in/app/services/v2/chip/service.rb]: As synced offline, added a new method for fixing rubocop failure.  For now disabled rubocop rule as suggested.  Pls find the changes in [commit](https://github.com/department-of-veterans-affairs/vets-api/pull/9336/commits/604e4613bd3bd2d414058dde830ed8cb55b7688f)

**Review by randomsync — APPROVED** [review]
lgtm

**Review by rileyanderson — APPROVED** [review]

**Pull request merged**